src/broadcast/filelistener.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <QFile>
+#include "broadcast/scrobblingservice.h"
+
+class FileListener: public ScrobblingService {
+  public:
+    FileListener(const QString &path);
+    void broadcastCurrentTrack(TrackPointer pTrack) override;
```

Comment:
Inheritance is unnecessary. The file listener could be implemented as a standalone class with 2 slots, connected to the ScrobblingService.

Reply:
The idea is to have N services that connect to the scrobbling device that answer to both slots. If I take one implementation out of the hierarchy then I have to update that separately. It also doesn't allow me to mock up the file listener. I don't see any advantage on breaking the inheritance. Can you elaborate on this?

Reply:
The scrobbling manager can offer 2 public signals for this purpose and dependent services can connect to them.

Reply:
Yeah but who owns the services? I was thinking of getting a list of services from the user settings, pass them to the scobbling manager and then simply keep them updated. I find the manager owning a list of services to be easier to mantain than creating a connection for every service and storing them somewhere else.